native/core/src/common/bit.rs

```diff
@@ -553,8 +564,11 @@ pub struct BitReader {
 /// either byte aligned or not.
 impl BitReader {
     pub fn new(buf: Buffer, len: usize) -> Self {
-        let num_bytes = cmp::min(8, len);
-        let buffered_values = read_num_bytes_u64(num_bytes, buf.as_slice());
+        let buffered_values = if 8 > len {
```

Comment:
nit: should we use size_of instead of 8?

Reply:
updated